core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -330,7 +351,9 @@ private[spark] object Utils extends Logging {
   def createTempDir(
       root: String = System.getProperty("java.io.tmpdir"),
       namePrefix: String = "spark"): File = {
-    JavaUtils.createTempDir(root, namePrefix)
+    val dir = createDirectory(root, namePrefix)
```

Comment:
https://github.com/apache/spark/pull/36529#discussion_r1153918687

@srowen @sadikovi 

After this revert, they are two different implementations  I think this can solve the problem. For `JavaUtils.createTempDir`, I will find other ways to solve it, it is only call by test code now